lib/ContainerManager.js

```diff
@@ -146,6 +151,32 @@ Server.prototype.configure = function(config, done) {
   done();
 };
 
+Server.prototype.jsonWebTokenParser = function(publicKey) {
+  var handleError = function(res) {
+    res.status(401);
+    res.json({error: 'Invalid JWT provided'});
+  };
+  return function(req, res, next) {
+    req.log.info(req.headers);
+    if (req.headers && req.headers.authorization) {
+      var parts = req.headers.authorization.split(' ');
+      if (parts.length === 2 && parts[0] === 'Bearer') {
+        var token = parts[1];
+      }
+      else {
+        handleError(res);
+      }
+    }
+    jwt.verify(token, publicKey, {algorithms: ['RS256']}, function(error, data) {
```

Comment:
Can we offer a config setting to pick the algorithm?

Reply:
Agreed on this. No need to do configurable options on this granular of a level. Obfuscates things too much.